godot/src/helpers_components/avatar_modifier_area_detector.gd

```diff
@@ -0,0 +1,50 @@
+extends Area3D
+
+@export var avatar: Avatar = null
+signal set_avatar_modifier_area(area: DclAvatarModifierArea3D)
+# Logic explanation:
+# If the avatar enters/exits an AREA but it is not in the scene
+# we don't check if it's inside or not...
+# and when the avatar enters/exits the scene, it will check it
+
+signal unset_avatar_modifier_area
+
+var overlapping_areas: Array[Area3D] = []
+
+
+func _ready():
+	avatar.change_scene_id.connect(self._on_avatar_change_scene_id)
```

Comment:
not sure if the connect link with a weak reference, could we check if it does?
otherwise I'd disconnect the signal in a eventual `tree_exited/exiting`